DataSynapseApi/ArmoniK.DevelopmentKit.GridServer.Client/Service.cs

```diff
@@ -288,33 +287,26 @@ private byte[] ActiveGetResult(params string[] taskId)
       };
       var idx = 0;
 
-      Logger.BeginPropertyScope(("SessionId", SessionId), ("Function", "ActiveGetResults"));
+      Logger.BeginPropertyScope(("SessionId", SessionId),
+                                ("Function", "ActiveGetResults"));
 
       while (missing.Count != 0)
       {
-        var buckets = missing.Batch(10000).ToList();
-
-        buckets.ForEach(bucket =>
+        foreach (var bucket in missing.Batch(10000))
         {
           var partialResults = SessionService.TryGetResults(bucket);
```

Comment:
`.TryGetResults` should probably return an `IList` instead of an `IEnumerable` to guarantee that you will be able to iterate multiple times.

By the way, `.TryGetResults` have this in the return `.ToList().Where(el => el != null);`, but should most likely be like this:
`.Where(el => el != null).ToList()`.